cms/apps/pages/admin.py

```diff
@@ -294,7 +294,7 @@ def get_form(self, request, obj=None, **kwargs):
             invalid_parents.add(obj.id)
         else:
             invalid_parents = frozenset()
-        homepage = request.pages.homepage
+        homepage = Page.objects.filter(parent=None).first()
```

Comment:
Any benefit to adding this as a cached property on the class since it's referenced twice?

Reply:
Not sure yet, going to do some testing. Though given that it's the exact same query, cachalot should pick it up.